eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -0,0 +1,237 @@
+            pool:
+              vmImage: ubuntu-16.04
+
+            steps:
+              - task: UsePythonVersion@0
+                displayName: 'Use Python 3.6'
+                inputs:
+                  versionSpec: '3.6'
+
+              - script: |
+                  python --version
+                  python eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --vv --ai ${{artifact.name}} --gi ${{artifact.groupId}}
+                displayName: 'Verify version for release'
+
+              - pwsh: |
+                  $(Build.SourcesDirectory)/eng/versioning/scan_for_unreleased_dependencies.ps1 ${{artifact.groupId}} ${{artifact.name}} $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                displayName: "Verify no unreleased dependencies"
+
+
+          - deployment: TagRepository
+            displayName: "Create release tag"
+            condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
+            environment: github
+            timeoutInMinutes: 5
+            dependsOn: VerifyReleaseVersion
+
+            pool:
+              vmImage: vs2017-win2016
+
+            strategy:
+              runOnce:
+                deploy:
+                  steps:
+                    - checkout: self
+                    - pwsh: |
+                        Get-ChildItem $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                        New-Item -Type Directory -Name ${{artifact.safeName}} -Path $(Pipeline.Workspace)
+                        Copy-Item $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.name}}-[0-9]*.[0-9]*.[0-9]* $(Pipeline.Workspace)/${{artifact.safeName}}
+                        Get-ChildItem $(Pipeline.Workspace)/${{artifact.safeName}}
+                      workingDirectory: $(Pipeline.Workspace)
+                      displayName: Stage artifacts
+                      timeoutInMinutes: 5
+                    - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
+                      parameters:
+                        ArtifactLocation: $(Pipeline.Workspace)/${{artifact.safeName}}
+                        PackageRepository: Maven
+                        ReleaseSha: $(Build.SourceVersion)
+                        RepoId: Azure/azure-sdk-for-java
+
+          - ${{if ne(artifact.options.skipPublishPackage, 'true')}}:
+            - deployment: PublishPackage
+              displayName: "Publish to Maven Central"
+              condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
+              environment: maven
+              dependsOn: TagRepository
+
+              pool:
+                vmImage: windows-2019
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                      - checkout: azure-sdk-build-tools
+                        path: azure-sdk-build-tools
```

Comment:
Do you need to specify the path? I think it is redundant.

Reply:
It is required to because it is used to reference other tools called by java-publishing.yml. e.g [here](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools?path=%2Ftools%2Fjava-publishing%2Fjava-publishing.yml&version=GBmaster&line=10&lineEnd=11&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)

Reply:
I'm Ok with it being here but doesn't it already default to that value?

Reply:
Also we should consider putting that in one of the azure-sdk-build-tools templates or at least pass the location as a template parameter as we don't want to have these hidden contracts about where things need to be setup.

Reply:
The default checkout path is usually just `s` so it still needs to be specified.
I moved `java-publishing.yml` to the Java Repo and passed the buildTools path as a parameter.

Reply:
OK I guess it only adds the repo path if it detects multiple checkouts.